File: src/features/typing/data/java/pascals_triangle.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<>();
            for (int j = 0; j <= i; j++) {
                row.add(0);
            }
            row.set(0, 1);
            row.set(i, 1);
            for (int j = 1; j < i; j++) {
                row.set(j, result.get(i - 1).get(j - 1) + result.get(i - 1).get(j));
            }
            result.add(row);
        }
        return result;
    }
}

File: src/features/typing/data/dart/pascals_triangle.dart
class Solution {
  List<List<int>> generate(int numRows) {
    final result = <List<int>>[];
    for (int i = 0; i < numRows; i++) {
      final row = List<int>.filled(i + 1, 0);
      row[0] = 1;
      row[i] = 1;
      for (int j = 1; j < i; j++) {
        row[j] = result[i - 1][j - 1] + result[i - 1][j];
      }
      result.add(row);
    }
    return result;
  }
}
